### ISSUE TITLE
Improve support for Django with custom uwsgi function

### ISSUE THREAD

**Issue by bxm156** [Miserlou/Zappa#1164]
## Context
I use django, but prefer to use my own uwsgi function so that I can wrap certain middlewares for monitoring. Zappa is great because I just use ```app_function```, instead of ```django_settings```, and viola, a django app running with some custom uwsgi middleware. However, this prevents me from using certain zappa related django commands, like ```zappa manage```. The handler only allows them to be used if django_settings is defined in the configuration, but the existance of ```django_settings``` overrides the use of app_function.

## Expected Behavior
Currently, when i deploy my app, I have to ensure ```djanog_settings``` is absent, and app_function is present. When I want to use the ```zappa manage``` command, I just edit my local config to use ```django_settings```, instead of ```app_function``` (Note: I don't deploy that change).

## Actual Behavior
I think a little refactoring in the handler would make this a lot easier.  We could refactor the handler to consider ```app_function``` the main/default function to use by zappa. ```django_settings``` is only used if ```app_function``` is not present. This allows me to utilize both in my configuration, so that I can use both the ```app_function``` in lambda, while still allowing the use of ```zappa manage```

## Possible Fix
I wanted to open this ticket to open a discussion on making ```app_function``` the preferred choice by Zappa,  and only defaulting to other things like the ```django_settings``` if it doesn't exist.

**Comment by aarcro** [comment]
I need to support [django-configurations](https://github.com/jazzband/django-configurations)

**Comment by hughgrigg** [comment]
@aarcro Were you able to get Zappa working with django-configurations?

**Comment by aarcro** [comment]
@hughgrigg Yes, take a look at #1251 It lets you use Django, but with a custom wsgi file. I've used it for both django-configurations and new-relic.
